File: Programmers/pro_43163_단어변환.py
from collections import deque,defaultdict
visited=defaultdict(int)

def solution(begin, target, words):    
    que = deque()
    que.append(begin)
    answer = -1
    while que:
        answer+=1
        for i in range(len(que)):
            cur = que.popleft()
            visited[cur]=1
            if(cur==target):
                return answer
            for word in words:
                if(visited[word]==0):
                    diff=0
                    for a,b in zip(cur,word):
                        if(a!=b):
                            diff+=1
                        if(diff>1):
                            break
                    if(diff==1):
                        que.append(word)        
    return 0

File: Programmers/pro_43163_단어변환.js
function solution(begin, target, words) {
  const visited = {}
  const dfs = (word, cnt) => {
    if (word === target) return cnt - 1
    let answer = 0
    visited[word] = 1
    const candidates = words.filter((s) => !visited[s] && [...s].filter((_, i) => s[i] === word[i]).length === begin.length - 1)
    for (let i = 0, l = candidates.length; i < l; i++) {
      const ret = dfs(candidates[i], cnt + 1)
      if (ret) answer = ret
    }
    return answer
  }
  return dfs(begin, 1);
}